desertbot/modules/commands/RSS.py

```diff
@@ -0,0 +1,189 @@
+            else:
+                return ("{!r} is not a valid subcommand, use {}help rss for a list of subcommands"
+                        .format(query[1], self.bot.commandChar))
+
+    def checkFeeds(self, message: IRCMessage):
+        if message.command in self.triggers():
+            if message.parameterList[0].lower() == "follow":
+                return self._followFeed(message)
+            elif message.parameterList[0].lower() == "unfollow":
+                return self._unfollowFeed(message)
+            elif message.parameterList[0].lower() == "toggle":
+                return self._toggleFeedSuppress(message)
+            elif message.parameterList[0].lower() == "list":
+                return self._listFeeds(message)
+            elif len(message.parameters.strip()) > 0:
+                feed = message.parameters.strip()
+                latest = self._getLatest(feed)
+                if latest is not None:
+                    response = 'Latest {}: {} | {}'.format(latest["name"], latest["title"], latest["link"])
+                    return IRCResponse(ResponseType.Say, response, message.replyTo)
+                else:
+                    return IRCResponse(ResponseType.Say,
+                                       "{} is not an RSS feed I monitor, leave a tell if you'd like it added!".format(
+                                           message.parameters.strip()),
+                                       message.replyTo)
+            else:
+                return self.help(None)
+
+        responses = []
+        for feedName, feedDeets in self.feeds.items():
+            if feedDeets["lastCheck"] > (datetime.datetime.utcnow() - datetime.timedelta(minutes=10)).isoformat():
+                continue
+
+            self.feeds[feedName]["lastCheck"] = datetime.datetime.utcnow().isoformat()
+
+            response = self.bot.moduleHandler.runActionUntilValue("fetch-url", feedDeets["url"])
+
+            if not response:
+                self.logger.warning("failed to fetch {!r}, either a server hiccup "
+                                    "or the feed no longer exists".format(feedDeets["url"]))
+                continue
+
+            soup = BeautifulSoup(response.content, "lxml")
+            item = soup.find("item")
+
+            if item is None:
+                self.logger.warning("the feed at {!r} doesn't have any items, has it shut down?"
+                                    .format(feedDeets["url"]))
+                continue
+
+            itemDate = item.find("pubdate").text
+            newestDate = dparser.parse(itemDate, fuzzy=True, ignoretz=True).isoformat()
+
+            if newestDate > feedDeets["lastUpdate"]:
+                self.feeds[feedName]["lastUpdate"] = newestDate
+                title = item.find("title").text
+                link = item.find("link").text
+                link = self.bot.moduleHandler.runActionUntilValue("shorten-url", link)
```

Comment:
This should probably fall back on the long link if URL shortening fails for whatever reason.